commands/create_cluster_test.go

```diff
@@ -76,3 +96,47 @@ func Test_CreateFromBadYAML01(t *testing.T) {
 		t.Error("Expected owner to be empty, got: ", myDef.Owner)
 	}
 }
+
+// Test_CreateFromCommandLine tests a cluster creation completely based on
+// command line arguments
+func Test_CreateFromCommandLine(t *testing.T) {
+	// mock server always responding positively
+	mockServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		fmt.Printf("mockServer request: %s %s\n", r.Method, r.URL)
+		w.Header().Set("Content-Type", "application/json")
+		w.Header().Set("Location", "/v4/clusters/f6e8r/")
+		w.WriteHeader(http.StatusCreated)
+		w.Write([]byte(`{"code": "RESOURCE_CREATED", "message": "Yeah!"}`))
+
+	}))
+	defer mockServer.Close()
+
+	// first run, with minimal arguments
+	args := []string{}
+	cmdAPIEndpoint = mockServer.URL
+	cmdOwner = "acme"
+	checkAddCluster(CreateClusterCommand, args)
+	addCluster(CreateClusterCommand, args)
+
+	// second run, with additional arguments
+	cmdClusterName = "UnitTestCluster"
```

Comment:
@marians cool thanks that will help. I get know that you're setting the command line vars. I was thinking they'd be set via Cobra or some similar library. 

Thanks for the clarification. I had some gaps in my knowledge of how gsctl works :)